applications/FSIApplication/python_scripts/fsi_coupling_interface.py

```diff
@@ -154,9 +340,30 @@ def _get_partitioned_fsi_utilities(self):
 
     def _create_partitioned_fsi_utilities(self):
         domain_size = self.GetFatherModelPart().ProcessInfo[KratosMultiphysics.DOMAIN_SIZE]
-        if domain_size == 2:
-            return KratosFSI.PartitionedFSIUtilitiesArray2D()
-        elif domain_size == 3:
-            return KratosFSI.PartitionedFSIUtilitiesArray3D()
+        if not self.GetFatherModelPart().IsDistributed():
+            if domain_size == 2:
+                if self.scalar_output:
+                    return KratosFSI.PartitionedFSIUtilitiesDouble2D()
+                else:
+                    return KratosFSI.PartitionedFSIUtilitiesArray2D()
+            elif domain_size == 3:
+                if self.scalar_output:
+                    return KratosFSI.PartitionedFSIUtilitiesDouble3D()
+                else:
+                    return KratosFSI.PartitionedFSIUtilitiesArray3D()
+            else:
+                raise Exception("Domain size expected to be 2 or 3. Got " + str(self.domain_size))
         else:
-            raise Exception("Domain size expected to be 2 or 3. Got " + str(self.domain_size))
+            self._epetra_communicator = KratosTrilinos.CreateCommunicator()
+            if domain_size == 2:
+                if self.scalar_output:
+                    return KratosTrilinos.TrilinosPartitionedFSIUtilitiesDouble2D(self._epetra_communicator)
+                else:
+                    return KratosTrilinos.TrilinosPartitionedFSIUtilitiesArray2D(self._epetra_communicator)
+            elif domain_size == 3:
+                if self.scalar_output:
+                    return KratosTrilinos.TrilinosPartitionedFSIUtilitiesDouble3D(self._epetra_communicator)
+                else:
+                    return KratosTrilinos.TrilinosPartitionedFSIUtilitiesArray3D(self._epetra_communicator)
```

Comment:
I think it is time to create a TrilinosExtension for the FSIApp
it is pretty straight forward (ofc not in this PR)

Reply:
Totally agree. Indeed I think is the only app missing. This is in top of my TODO lists now.